src/main/java/frc/auto/AutoOperate.java

```diff
@@ -55,13 +56,19 @@ private void driveMode() {
         currentTime = System.currentTimeMillis() - startTime;
 
         try {
-            if(currentTime >= data.get(0).get(0).longValue()) {
+            if(currentTime < data.get(0).get(0).longValue()) {
                 this.drive.setDriveLeft(data.get(0).get(1));
                 this.drive.setDriveRight(data.get(0).get(2));
+            } else {
                 data.remove(0);
             }
+            if(data.size() <= 0) {
+                this.drive.resetPosition();
+                this.disable();
+                return;
+            }
         } catch (Exception e) {
-            this.drive.disable();
+            this.disable();
```

Comment:
If the code fails then the robot just stops, maybe you should make a System.out.err message with some context of where it failed, and later we may add this into a log file

Reply:
Agreed, will change that to a System.err.println();

Reply:
Agreed, will change that to System.err.println();